src/core/Directus/Services/ItemsService.php

```diff
@@ -230,43 +232,52 @@ public function validateManyToManyCollection($payload, $params, $aliasColumnDeta
             $parentCollectionPrimaryKey = SchemaService::getCollectionPrimaryKey($parentCollectionName);
             $relationalCollectionColumns = SchemaService::getAllCollectionFields($relationalCollectionName);
             $foreignJoinColumn = $aliasColumnDetails->getRelationship()->getFieldMany();
 
-            foreach($payload[$colName] as $individual){
-
-                if(!isset($individual['$delete'])){
-                    foreach($relationalCollectionColumns as $key => $column){
-                        if(!$column->hasPrimaryKey() && !empty($individual[$relationalCollectionPrimaryKey])){
-                            $columnName = $column->getName();
-                            $relationalCollectionData = $this->findByIds(
-                                $relationalCollectionName,
-                                $individual[$relationalCollectionPrimaryKey],
-                                $params
-                            );
-                            $individual[$columnName] = array_key_exists($columnName, $individual) ? $individual[$columnName]: (isset($relationalCollectionData['data'][$columnName]) ? ((DataTypes::isJson($column->getType()) ? (array) $relationalCollectionData['data'][$columnName] : $relationalCollectionData['data'][$columnName])) : null);
+            // Merge the payload with the DB object if it's getting updated or created and for M2O only ID will be passed. Thus no need to merge it with the DB object.
+            if (is_array($payload[$colName])) {
+                foreach ($payload[$colName] as $individual) {
+                    if (!isset($individual['$delete'])) {
+                        foreach ($relationalCollectionColumns as $key => $column) {
+                            if (!$column->hasPrimaryKey() && !empty($individual[$relationalCollectionPrimaryKey])) {
+                                $columnName = $column->getName();
+                                $relationalCollectionData = $this->findByIds(
+                                    $relationalCollectionName,
+                                    $individual[$relationalCollectionPrimaryKey],
+                                    $params
+                                );
+                                $individual[$columnName] = array_key_exists($columnName, $individual) ? $individual[$columnName] : (isset($relationalCollectionData['data'][$columnName]) ? ((DataTypes::isJson($column->getType()) ? (array) $relationalCollectionData['data'][$columnName] : $relationalCollectionData['data'][$columnName])) : null);
+                            }
+                        }
+                        // only add parent id's to items that are lacking the parent column
+                        if (empty($individual[$foreignJoinColumn])) {
+                            $individual[$foreignJoinColumn] = !empty($recordData[$parentCollectionPrimaryKey]) ? $recordData[$parentCollectionPrimaryKey] : 0;
```

Comment:
It defaults to `0`? What's going on here?

Reply:
It's not added in this PR. But still let me try to explain the code - based on what I understood.

This'll add a foreign key into O2M collection to boycott the validation.

```

1. **Post**
   id
   description
   comment - O2M

2. **Comment**
    id
    description
    post - M2O

```

Let's assume comment have a required `post` field. 

If user try to add `comment` from `post` collection;  user will not select the `post` from that `comment` popup. Thus there's chance that the `post` will be null or will not be passed in a payload.  At that time the required validation will be throw the error and to handle this it's been set temporarily.

I verified that the DB not contain the 0 as a value.